Tests/Unit/CommonResourceHelper.Tests.ps1

```diff
@@ -0,0 +1,181 @@
+﻿$errorActionPreference = 'Stop'
+Set-StrictMode -Version 'Latest'
+
+Describe 'CommonResourceHelper Unit Tests' {
+    BeforeAll {
+        # Import the CommonResourceHelper module to test
+        $testsFolderFilePath = Split-Path -Path $PSScriptRoot -Parent
+        $moduleRootFilePath = Split-Path -Path $testsFolderFilePath -Parent
+        $dscResourcesFolderFilePath = Join-Path -Path $moduleRootFilePath -ChildPath 'DscResources'
+        $commonResourceHelperFilePath = Join-Path -Path $dscResourcesFolderFilePath -ChildPath 'CommonResourceHelper.psm1'
+        Import-Module -Name $commonResourceHelperFilePath
+    }
+
+    InModuleScope 'CommonResourceHelper' {
+        Describe 'Test-IsNanoServer' {
+            $testComputerInfoNanoServer = @{
+                OsProductType = 'Server'
+                OsServerLevel = 'NanoServer'
+            }
+
+            $testComputerInfoServerNotNano = @{
+                OsProductType = 'Server'
+                OsServerLevel = 'NotNano'
+            }
+
+            $testComputerInfoNotServer = @{
+                OsProductType = 'NotServer'
+                OsServerLevel = 'NotNano'
+            }
+
+            Mock -CommandName 'Test-CommandExists' -MockWith { return $true }
+            Mock -CommandName 'Get-ComputerInfo' -MockWith { return $testComputerInfoNanoServer }
+
+            Context 'Get-ComputerInfo command exists' {
+                Context 'Computer OS type is Server and OS server level is NanoServer' {
+                    It 'Should not throw' {
+                        { $null = Test-IsNanoServer } | Should Not Throw
+                    }
+
+                    It 'Should test if the Get-ComputerInfo command exists' {
+                        $testCommandExistsParameterFilter = {
+                            $nameParameterCorrect = $Name -eq 'Get-ComputerInfo'
+                            return $nameParameterCorrect
+                        }
+
+                        Assert-MockCalled -CommandName 'Test-CommandExists' -ParameterFilter $testCommandExistsParameterFilter -Exactly 1 -Scope 'Context'
+                    }
+
+                    It 'Should retrieve the computer info' {
+                        Assert-MockCalled -CommandName 'Get-ComputerInfo' -Exactly 1 -Scope 'Context'
+                    }
+
+                    It 'Should return true' {
+                        Test-IsNanoServer | Should Be $true
+                    }
+                }
+
+                Mock -CommandName 'Get-ComputerInfo' -MockWith { return $testComputerInfoServerNotNano }
```

Comment:
Why are you defining the Mock at the describe level multiple times?  Can this be moved into the context?  It may not be obvious to others you did not clean this up. also done in other places.